scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala

```diff
@@ -296,6 +296,29 @@ object TreeSyntax {
       }
       !isOnlyChildOfOnlyChild(t)
     }
+    @annotation.tailrec
+    def guessNeedsLineSep(t: Tree): Boolean = t match {
+      case _: Decl.Val | _: Decl.Var | _: Decl.Def | _: Defn.Macro | _: Defn.Type | _: Term.This |
+          _: Term.Name | _: Term.Select | _: Member.Apply | _: Term.Ascribe | _: Term.Tuple |
```

Comment:
please add `Term.Ref`, and remove its children `Term.This`, `Term.Name`, `Term.Select`, while moving up `Term.ApplyUnary` from further down.